pkg/cli/image/manifest/manifest.go

```diff
@@ -51,7 +51,7 @@ type SecurityOptions struct {
 }
 
 func (o *SecurityOptions) Bind(flags *pflag.FlagSet) {
-	flags.StringVarP(&o.RegistryConfig, "registry-config", "a", o.RegistryConfig, "Path to your registry credentials (defaults to ~/.docker/config.json)")
+	flags.StringVarP(&o.RegistryConfig, "registry-config", "a", o.RegistryConfig, "Path to your registry credentials. Alternatively env variable ${REGISTRY_AUTH_FILE} can be also specified. Defaults to ${XDG_RUNTIME_DIR}/containers/auth.json, ${XDG_CONFIG_HOME}/containers/auth.json, /run/containers/${UID}/auth.json, ${DOCKER_CONFIG}, ~/.docker/config.json, ~/.dockercfg")
```

Comment:
We can't change the order, since that might break current users. We'll have to do it gradually:
1. Add podman locations after `.docker/config.json`, we talked with @atiratree about it, it'll be done by re-using the current logic to read `.docker/config.json` and add everything else using `github.com/containers/image` library. We just need to ensure that the former has higher precedence than the latter.
2. Add a warning when using the old loading order.
3. Add support for the new order, but behind an env var, `CONTAINERS_AUTH_ORDER` or something like that. This would basically drop the legacy support we currently have for `.docker/config.json` and just use `github.com/containers/image/`. This would apply both to reading and writing as well.

Don't forget:
1. Add release notes about the loading order changes including the env var to enable it  - `CONTAINERS_AUTH_ORDER`. Mention supporting `REGISTRY_AUTH_FILE` as well.
Call out that in 2 release, we'll flip `CONTAINERS_AUTH_ORDER` to on by default.

Reply:
@soltysh I think I have resolved all of the issues. Can you please rereview?

regarding the release notes, I will write them just after we confirm the name of the env variable